src/main/java/seedu/ddd/model/contact/common/Contact.java

```diff
@@ -120,7 +120,8 @@ public boolean isSameContact(Contact otherContact) {
         }
 
         return otherContact != null
-                && otherContact.getName().equals(getName());
+                && (otherContact.getName().isSameName(name)
+                || otherContact.getPhone().equals(phone));
```

Comment:
Should this be changed to && instead?

Reply:
If you use &&, same phone and different name or same name but different phone will be treated as different contact. But I thought if either one is the same, the contact should be considered to be the same?